packages/superset-ui-dimension/src/parseLength.ts

```diff
@@ -0,0 +1,17 @@
+const HUNDRED_PERCENT = { isDynamic: true, multiplier: 1 } as const;
+
+const pxPattern = /px$/;
+
+export default function parseLength(
+  input: string | number,
+): { isDynamic: true; multiplier: number } | { isDynamic: false; value: number } {
+  if (input === 'auto' || input === '100%') {
+    return HUNDRED_PERCENT;
+  } else if (typeof input === 'string' && input.length > 0 && input[input.length - 1] === '%') {
+    // eslint-disable-next-line no-magic-numbers
+    return { isDynamic: true, multiplier: parseFloat(input.substring(0, input.length - 1)) / 100 };
```

Comment:
you could do parseFloat(input). If the string starts with a number, parseFloat will just ignore the random characters. Comment applies below too.

Reply:
Oh cool. Didn't know that!